### PR TITLE
Return empty Response for empty Request, avoiding instantiating drivers

### PR DESCRIPTION
**Pull request by juanjux**
Part of src-d/backlog#889

### CODE REVIEW

**Comment by juanjux** [comment]
Yes, definitely, I'll update the docs. Another option would be to add a new response status `empty` or `warning` (the latter using `[]errors` to include the warning).

**Comment by juanjux** [comment]
Don't merge yet - I wan't to explore options to return an empty `Response.UAST` field instead of `nil`.

**Comment by bzz** [comment]
Just 5c - it would if that would have no implications on error handling for the clients

**Comment by juanjux** [comment]
I changed the response to have an empty `Node` in its `ParseResponse.UAST` field instead of `nil`. This should make it less prone to cause unwanted `nil` dereferences in user code (even trough I think users should check any pointers against `nil` before dereferencing, in general) and have a better correspondence (empty code -> empty UAST).

PTAL and merge if approved.

File: server.go [hunk 1]
```diff
@@ -119,6 +120,15 @@ func (s *Server) Parse(req *protocol.ParseRequest) *protocol.ParseResponse {
 		lang = GetLanguage(req.Filename, []byte(req.Content))
 	}
 
+	// If the code Content is empty, just return an empty reponse
+	if req.Content == "" {
+		logrus.Info("Empty code received, returning empty UAST")
```
Comment (abeaumont) [inline server.go]: Not very relevant information about the server operation itself, maybe a `Debug` is better?

**Pull request merged**